diff --git a/app/protocol_filter.node.ts b/app/protocol_filter.node.ts
index 483d1c4e4..66609d9e9 100644
--- a/app/protocol_filter.node.ts
+++ b/app/protocol_filter.node.ts
@@ -68,6 +68,7 @@ export function _createFileHandler({
   const allowedRoots = [
     userDataPath,
     installPath,
+    '@noto-emoji-pngs@',
     getAvatarsPath(userDataPath),
     getBadgesPath(userDataPath),
     getDraftPath(userDataPath),
diff --git a/ts/components/fun/FunEmoji.dom.tsx b/ts/components/fun/FunEmoji.dom.tsx
index ea031e9a8..3d760a43e 100644
--- a/ts/components/fun/FunEmoji.dom.tsx
+++ b/ts/components/fun/FunEmoji.dom.tsx
@@ -3,7 +3,6 @@
 import classNames from 'classnames';
 import type { CSSProperties, JSX } from 'react';
 import { useMemo, useState, useCallback } from 'react';
-import MANIFEST from '../../../build/jumbomoji.json';
 import type { FunImageAriaProps } from './types.dom.tsx';
 import { createLogger } from '../../logging/log.std.ts';
 import { Emoji } from '../../axo/emoji.std.ts';
@@ -24,20 +23,17 @@
 const FUN_INLINE_EMOJI_SMALL_CLASS = 'FunInlineEmoji__Small';
 const FUN_INLINE_EMOJI_JUMBO_CLASS = 'FunInlineEmoji__Jumbo';
 
-const KNOWN_JUMBOMOJI = new Set<string>(Object.values(MANIFEST).flat());
-const MIN_JUMBOMOJI_SIZE = 33;
-
 function getEmojiJumboUrl(
   emoji: Emoji.Variant,
-  size: number | undefined
+  _size: number | undefined
 ): string | null {
-  if (size != null && size < MIN_JUMBOMOJI_SIZE) {
-    return null;
-  }
-  if (KNOWN_JUMBOMOJI.has(emoji)) {
-    return `emoji://jumbo?emoji=${encodeURIComponent(emoji)}`;
-  }
-  return null;
+  const emojiToNotoName = (emojiV: string): string =>
+    // oxlint-disable-next-line typescript-eslint(no-misused-spread)
+    `emoji_u${[...emojiV]
+      .filter(c => c !== '\ufe0f')
+      .map(c => c.codePointAt(0)?.toString(16).padStart(4, '0'))
+      .join('_')}.png`;
+  return `file://@noto-emoji-pngs@/${emojiToNotoName(emoji)}`;
 }
 
 export type FunStaticEmojiSize =
